src/constraints.jl

```diff
@@ -141,20 +131,24 @@ end
     Get constraint by name
 =#
 
-# this covers the non-type-stable get(m, ConstraintIndex) case
-function MOI.canget(model::LinQuadOptimizer, ::Type{MOI.ConstraintIndex}, name::String)
-    return haskey(model.constraint_names_rev, name)
-end
-function MOI.get(model::LinQuadOptimizer, ::Type{<:MOI.ConstraintIndex}, name::String)
-    return model.constraint_names_rev[name]
+function MOI.get(model::LinQuadOptimizer, ::Type{<:MOI.ConstraintIndex}, 
+                 name::String)
+    if haskey(model.constraint_names_rev, name)
+        return model.constraint_names_rev[name]
+    else
+        return nothing
```

Comment:
I'm not sure that it should return `nothing`. Shouldn't it throw an error?

Reply:
It follows MOI requirements:
https://github.com/JuliaOpt/MathOptInterface.jl/blob/master/src/attributes.jl#L160-L163

Whether it's good or not is another question, and should be tackled on MOI level.